src/groupByKey.ts

```diff
@@ -1,7 +1,22 @@
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [keyName: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): object {
   // write code here;
+  const result: GroupsMap<T> = {};
+
+  items.forEach((itemsElement: T) => {
+    const variable: string = String(itemsElement[key]);
```

Comment:
it's not about the type of variable but about its name. You shouldn't name variables "variable". It doesn't give any new information. Name it according to what it stores

Reply:
A, okay)